ccl/sqlproxyccl/acl: TestParsingErrorHandling failed

ccl/sqlproxyccl/acl.TestParsingErrorHandling [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9113225?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9113225?buildTab=artifacts#/) on master @ [11fbbca584e3bd774ebee3ee94c49ca30a096b26](https://github.com/cockroachdb/cockroach/commits/11fbbca584e3bd774ebee3ee94c49ca30a096b26):


```
=== RUN   TestParsingErrorHandling
--- FAIL: TestParsingErrorHandling (0.88s)
=== RUN   TestParsingErrorHandling/error_after_update
    file_test.go:499: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/acl/file_test.go:499
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestParsingErrorHandling/error_after_update
E230317 06:41:02.334749 97 ccl/sqlproxyccl/acl/file.go:71  [-] 14  error updating access control list from file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling1756134841/001/allowlist.yaml modified at 2023-03-17 06:41:01.450593078 +0000 UTC: yaml: unmarshal errors:
E230317 06:41:02.334749 97 ccl/sqlproxyccl/acl/file.go:71  [-] 14 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230317 06:41:02.334948 97 ccl/sqlproxyccl/acl/file.go:110  [-] 15  Could not read file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling1756134841/001/allowlist.yaml for *acl.Allowlist: yaml: unmarshal errors:
E230317 06:41:02.334948 97 ccl/sqlproxyccl/acl/file.go:110  [-] 15 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230317 06:41:02.335006 97 ccl/sqlproxyccl/acl/file.go:100  [-] 16  WatchList daemon stopped: context canceled
    --- FAIL: TestParsingErrorHandling/error_after_update (0.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-sessions @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParsingErrorHandling.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25541

ccl/sqlproxyccl/acl.TestParsingErrorHandling [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9139159?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9139159?buildTab=artifacts#/) on master @ [9aa4f3485c75397e9ce0edabbf22c6343e1a901a](https://github.com/cockroachdb/cockroach/commits/9aa4f3485c75397e9ce0edabbf22c6343e1a901a):


```
=== RUN   TestParsingErrorHandling
--- FAIL: TestParsingErrorHandling (0.41s)
=== RUN   TestParsingErrorHandling/error_after_update
    file_test.go:499: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/acl/file_test.go:499
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestParsingErrorHandling/error_after_update
    --- FAIL: TestParsingErrorHandling/error_after_update (0.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParsingErrorHandling.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/sqlproxyccl/acl.TestParsingErrorHandling [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9151205?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9151205?buildTab=artifacts#/) on master @ [5839fe0fd13aa33f247e6591875e197fa451e770](https://github.com/cockroachdb/cockroach/commits/5839fe0fd13aa33f247e6591875e197fa451e770):


```
=== RUN   TestParsingErrorHandling
--- FAIL: TestParsingErrorHandling (1.21s)
=== RUN   TestParsingErrorHandling/error_after_update
E230320 16:17:42.842486 93 ccl/sqlproxyccl/acl/file.go:71  [-] 14  error updating access control list from file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling777176561/001/allowlist.yaml modified at 2023-03-20 16:17:42.705381065 +0000 UTC: yaml: unmarshal errors:
E230320 16:17:42.842486 93 ccl/sqlproxyccl/acl/file.go:71  [-] 14 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230320 16:17:42.842650 93 ccl/sqlproxyccl/acl/file.go:110  [-] 15  Could not read file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling777176561/001/allowlist.yaml for *acl.Allowlist: yaml: unmarshal errors:
E230320 16:17:42.842650 93 ccl/sqlproxyccl/acl/file.go:110  [-] 15 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230320 16:17:42.943698 93 ccl/sqlproxyccl/acl/file.go:71  [-] 16  error updating access control list from file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling777176561/001/allowlist.yaml modified at 2023-03-20 16:17:42.705381065 +0000 UTC: yaml: unmarshal errors:
E230320 16:17:42.943698 93 ccl/sqlproxyccl/acl/file.go:71  [-] 16 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230320 16:17:42.943796 93 ccl/sqlproxyccl/acl/file.go:110  [-] 17  Could not read file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling777176561/001/allowlist.yaml for *acl.Allowlist: yaml: unmarshal errors:
E230320 16:17:42.943796 93 ccl/sqlproxyccl/acl/file.go:110  [-] 17 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230320 16:17:43.044164 93 ccl/sqlproxyccl/acl/file.go:71  [-] 18  error updating access control list from file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling777176561/001/allowlist.yaml modified at 2023-03-20 16:17:42.705381065 +0000 UTC: yaml: unmarshal errors:
E230320 16:17:43.044164 93 ccl/sqlproxyccl/acl/file.go:71  [-] 18 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230320 16:17:43.044254 93 ccl/sqlproxyccl/acl/file.go:110  [-] 19  Could not read file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling777176561/001/allowlist.yaml for *acl.Allowlist: yaml: unmarshal errors:
E230320 16:17:43.044254 93 ccl/sqlproxyccl/acl/file.go:110  [-] 19 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230320 16:17:43.144585 93 ccl/sqlproxyccl/acl/file.go:71  [-] 20  error updating access control list from file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling777176561/001/allowlist.yaml modified at 2023-03-20 16:17:42.705381065 +0000 UTC: yaml: unmarshal errors:
E230320 16:17:43.144585 93 ccl/sqlproxyccl/acl/file.go:71  [-] 20 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230320 16:17:43.144689 93 ccl/sqlproxyccl/acl/file.go:110  [-] 21  Could not read file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling777176561/001/allowlist.yaml for *acl.Allowlist: yaml: unmarshal errors:
E230320 16:17:43.144689 93 ccl/sqlproxyccl/acl/file.go:110  [-] 21 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230320 16:17:43.245050 93 ccl/sqlproxyccl/acl/file.go:71  [-] 22  error updating access control list from file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling777176561/001/allowlist.yaml modified at 2023-03-20 16:17:42.705381065 +0000 UTC: yaml: unmarshal errors:
E230320 16:17:43.245050 93 ccl/sqlproxyccl/acl/file.go:71  [-] 22 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230320 16:17:43.245143 93 ccl/sqlproxyccl/acl/file.go:110  [-] 23  Could not read file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling777176561/001/allowlist.yaml for *acl.Allowlist: yaml: unmarshal errors:
E230320 16:17:43.245143 93 ccl/sqlproxyccl/acl/file.go:110  [-] 23 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230320 16:17:43.345413 93 ccl/sqlproxyccl/acl/file.go:71  [-] 24  error updating access control list from file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling777176561/001/allowlist.yaml modified at 2023-03-20 16:17:42.705381065 +0000 UTC: yaml: unmarshal errors:
E230320 16:17:43.345413 93 ccl/sqlproxyccl/acl/file.go:71  [-] 24 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
E230320 16:17:43.345522 93 ccl/sqlproxyccl/acl/file.go:110  [-] 25  Could not read file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling777176561/001/allowlist.yaml for *acl.Allowlist: yaml: unmarshal errors:
E230320 16:17:43.345522 93 ccl/sqlproxyccl/acl/file.go:110  [-] 25 +  line 1: cannot unmarshal !!str ``no long...`` into acl.AllowlistFile
    file_test.go:517: should have gotten a new controller
    --- FAIL: TestParsingErrorHandling/error_after_update (0.89s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParsingErrorHandling.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/sqlproxyccl/acl.TestParsingErrorHandling [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9161790?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9161790?buildTab=artifacts#/) on master @ [b806772ef6ee57b8ef40f8921ab66b3042136033](https://github.com/cockroachdb/cockroach/commits/b806772ef6ee57b8ef40f8921ab66b3042136033):


```
=== RUN   TestParsingErrorHandling
--- FAIL: TestParsingErrorHandling (0.55s)
=== RUN   TestParsingErrorHandling/error_after_update
    file_test.go:499: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/acl/file_test.go:499
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestParsingErrorHandling/error_after_update
    --- FAIL: TestParsingErrorHandling/error_after_update (0.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParsingErrorHandling.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/sqlproxyccl/acl.TestParsingErrorHandling [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9161790?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9161790?buildTab=artifacts#/) on master @ [b806772ef6ee57b8ef40f8921ab66b3042136033](https://github.com/cockroachdb/cockroach/commits/b806772ef6ee57b8ef40f8921ab66b3042136033):


```
=== RUN   TestParsingErrorHandling
--- FAIL: TestParsingErrorHandling (0.58s)
=== RUN   TestParsingErrorHandling/error_after_update
E230321 04:02:41.585340 86 ccl/sqlproxyccl/acl/file.go:71  [-] 13  error updating access control list from file /artifacts/tmp/_tmp/72c09b96526627cf742479f65b1af982/TestParsingErrorHandling3946405301/001/error_file.idk modified at 2023-03-21 04:02:41.582434682 +0000 UTC: yaml: unmarshal errors:
E230321 04:02:41.585340 86 ccl/sqlproxyccl/acl/file.go:71  [-] 13 +  line 1: cannot unmarshal !!str ``not yaml`` into acl.AllowlistFile
    file_test.go:499: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/sqlproxyccl/acl/file_test.go:499
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestParsingErrorHandling/error_after_update
    --- FAIL: TestParsingErrorHandling/error_after_update (0.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParsingErrorHandling.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/sqlproxyccl/acl/file_test.go
+++ b/pkg/ccl/sqlproxyccl/acl/file_test.go
@@ -494,7 +494,7 @@ func TestParsingErrorHandling(t *testing.T) {
 		select {
 		case <-next:
 			t.Error("should not have gotten a new controller")
-		case <-time.After(time.Millisecond * 200):
+		case <-time.After(time.Millisecond * 500):
 			// error count should go up
 			require.Equal(t, int64(1), errorCountMetric.Snapshot().Value())
 		}
@@ -513,7 +513,7 @@ func TestParsingErrorHandling(t *testing.T) {
 					},
 				},
 			}, allowlist.entries)
-		case <-time.After(time.Millisecond * 200):
+		case <-time.After(time.Millisecond * 500):
 			t.Error("should have gotten a new controller")
 		}
 	})

